apps/system/js/browser_context_menu.js

```diff
@@ -51,6 +50,8 @@
         return;
       }
 
+      SettingsListener.observe(PINNING_PREF, '', this._setPinning.bind(this));
+
```

Comment:
You can probably move this call right before the comment that says: // Notify the embedder we are handling ...

The reason for it is because we may do some early returns in various cases, and in this case I'm not sure it is useful to have the observer.